Update comments in setup_inc.h

PBfordev opened the pull request:
I have updated some comments but there are still many which I think should be updated but I am not sure how.

**WXWIN_COMPATIBILITY_3_0**
> Default is 1 but please try building your code with 0 as the default will
> change to 0 in the next version and the deprecated functions will disappear
> in the version after it completely.

We have already discussed this but I still believe the comment is incorrect. There is no doubt that the default value is not 1 but 0. Moreover, I do not think that the version part should be identical to that for `WXWIN_COMPATIBILITY_3_2`, as 3.0 compatibility should be abandoned one (stable) version before 3.2 compatibility is.

**wxUSE_EXTENDED_RTTI**
> Recommended setting: 0 (this is still work in progress...)

AFAIK, there has been no progress on this in many years and the feature was even considered for removal?

**wxUSE_COMPILER_TLS**
[The comment](https://github.com/wxWidgets/wxWidgets/blob/8a77fae68b8758bbdcd729e5680e0c4bd874bfb6/include/wx/setup_inc.h#L284) mentions issues with Windows XP but since wxWidgets 3.3 supports only Vista+, not only should the comment be changed but wx/msw/chkconf.h should be modified as well?

**wxUSE_INTL**
> i18n support: _() macro, wxLocale class. Requires wxTextFile.

Does it really require wxTextFile - I cannot see that?

**wxUSE_DIALUP_MANAGER**
> Requires wxUSE_DYNAMIC_LOADER.

This does not seem to be the case.

**wxUSE_DYNAMIC_LOADER**
> experimental, don't use for now

Is it still experimental? Is it even used in wxWidgets code? I found `dynload.h` included in several places (e.g.,` src\msw\gdiplus.cpp`) but AFAICT, the code uses just `wxDynamicLibrary` (`dynlib.h`) and nothing from `dynload.h`?

**wxUSE_XML**
> XML parsing classes. Note that their API will change in the future, so using wxXmlDocument and wxXmlNode in your app is not recommended.

Is the Note correct?

**wxUSE_GEOMETRY**
> Miscellaneous geometry code: needed for Canvas library

Affine transformation seem to be used by `wx{GC}DC`? See also` wxUSE_DC_TRANSFORM_MATRIX` remark below.

**wxUSE_FILEDLG**
> TODO: setting to choose the generic or native one

Remove?

**wxUSE_IPC, wxUSE_HELP, wxUSE_CONSTRAINTS, and others**
I wonder why the comments for these are written like that instead of above the define as usual?

**wxUSE_DC_TRANSFORM_MATRIX**
It seems this is in source code used for detecting availability of `wxAffineMatrix2D`, but `wxAffineMatrix2D` depends on `wxUSE_GEOMETRY`?

lanurmi committed:
Use wxChoice for month selection in wxGenericCalendarCtrl

For no obvious reason the generic calendar control used
a read-only wxComboBox until now.

While this is more or less identical to a wxChoice on MSW,
it is somewhat different on OSX and Gtk: Whereas a wxChoice
can be opened by clicking anywhere on it, a wxComboBox requires
clicking on the arrows on the right edge. A wxComboBox
allows selecting and copying the current value, but this is
probably not terribly useful in a calendar.

See #22853.

vadz committed:
Allow passing NULL to wxWindow::CreateUsingMSWClass()

Instead of calling GetMSWClassName() in wxWindow::Create() itself, just
use NULL there and interpret this as meaning to use the default wxWindow
class in CreateUsingMSWClass(), as it will allow it to distinguish
between creating generic windows and native controls.

No real changes yet.

vadz committed:
Enable double buffering for all generic windows in wxMSW

Turn on WS_EX_COMPOSITED for child windows, as this is not supposed to
result in any bad effects and double buffers drawing, avoiding flicker.

vadz committed:
Set wxALWAYS_NATIVE_DOUBLE_BUFFER to 1 to wxMSW too

Double buffering is always used in wxMSW after the last commit, so there
is no need to implement double buffering in our own code there any more.

This also means that only wxMotif and wxGTK1 don't have native double
buffering now, so flip the direction of the check and test for them only
(which will make removing support for them simpler).

vadz committed:
Don't bother calling SetDoubleBuffered(true) in wxComboCtrl

This is already the case by default now.

vadz committed:
Put wxStdioPipe helper in an anonymous namespace

No real changes, just some hygiene.

vadz committed:
Fix using addr2line under Linux

Pass the correct module path, i.e. the path to a shared library and not
the main executable, if this is where the address is located, and adjust
it by the module offset to make addr2line actually work for us instead
of just always returning "??:0".

vadz committed:
Remove unnecessary lines from wxStackWalker::InitFrames()

The "name" variable was already filled by ReadLine(), there is no need
to copy the buffer into it again.

vadz committed:
Fix possible problem with wrong stack frame line number

The comment sating that the line number would be left as 0 was wrong, as
the line number could keep its value from the previous loop iteration
instead.

Do reset it to 0 to ensure we don't reuse the wrong value.

vadz committed:
Fix debug message in wxStackWalker::InitFrames()

No real changes, just use "colon" instead of "semicolon" because this is
what we expect here and also remove the unnecessary wxT() and .c_str().

vadz committed:
Extract code for parsing addr2line output in a function

No changes yet, this is a pure refactoring.

This commit is best viewed with Git --color-moved option.

vadz committed:
Improve parsing of line numbers in addr2line output

Take into account possible presence of more output after them.

vadz committed:
Remove CI builds not using C++11

Also add MSVS 2019 and 2022 builds to AppVeyor.

vadz committed:
Always require C++11 for CMake builds

vadz committed:
Change package version in configure to 3.3.0

vadz committed:
Require C++11 in configure

vadz committed:
Run Ubuntu 18.04 builds in a container

Ubuntu 18.04 runner is being discontinued by GitHub, so switch the
builds using it to using a container.

We probably ought to actually run all of the builds in containers to
make them independent of changes to GitHub environment.

vadz committed:
Remove 20+ years old workaround for gcc under HP-UX

This is almost surely not necessary any longer as the version of gcc
that must have needed that is no longer supported.

vadz committed:
Move Cygwin __USE_W32_SOCKETS definition to wxMSW-specific file

Don't do this in wx/defs.h itself.

vadz committed:
Always use variadic macros and remove wxNO_VARIADIC_MACROS

Variadic macros are always available in C++11, which will be required
very soon, so don't bother testing for them and just always use them.

vadz committed:
Check that C++11 is available and remove tests for it

Don't bother checking for various C++11 features that are available in
all C++11 compilers.

Also assume that std::exception_ptr is available in all still supported
MinGW versions and remove checks for it too (see #16634).

Further simplifications remain possible, this is just the first step.

PBfordev committed:
Update comments in setup_inc.h

Update the outdated comments.

PBfordev committed:
Update comments in setup_inc.h

Update the outdated comments.

PBfordev commented on include/wx/setup_inc.h:
```diff
@@ -1120,16 +1116,16 @@
 #define wxUSE_ARTPROVIDER_TANGO 1
 
-// Hotkey support (currently Windows only)
+// Hotkey support (currently Windows and macOS only)
```
Moreover, the description in "wxUSE Preprocessor Symbols" in the docs should be updated and macOS added to the note in `wxWindow::UnRegisterHotKey()` docs.

csomor committed:
fixing native coordinate system within wxStaticBox on macOS

see @22837

csomor committed:
fixing native coordinate system within wxStaticBox on macOS

see #22837

vadz commented:
> I have updated some comments but there are still many which I think should be updated but I am not sure how.

Thanks for the updates, I don't have any comments concerning the changes so far, they should all be applied.

> **WXWIN_COMPATIBILITY_3_0**
> 
> > Default is 1 but please try building your code with 0 as the default will
> > change to 0 in the next version and the deprecated functions will disappear
> > in the version after it completely.
> 
> We have already discussed this but I still believe the comment is incorrect. There is no doubt that the default value is not 1 but 0. Moreover, I do not think that the version part should be identical to that for `WXWIN_COMPATIBILITY_3_2`, as 3.0 compatibility should be abandoned one (stable) version before 3.2 compatibility is.
 
Yes, this became outdated after 3.2 and wasn't fully updated. It should instead say something like this:
```cpp
// Default is 0, but you may still set it to 1 if you can't update the existing
// code relying on the deprecated functions. Please do consider updating it if
// at all possible instead of changing this to 1, however, as these functions
// will completely disappear in the next wxWidgets release.
#define WXWIN_COMPATIBILITY_3_0 0
```

> **wxUSE_EXTENDED_RTTI**
> 
> > Recommended setting: 0 (this is still work in progress...)
> 
> AFAIK, there has been no progress on this in many years and the feature was even considered for removal?

Yes, we should probably say
```cpp
// This feature is incompletely and experimental, please only enable it if
// you want to participate in its development.
//
// Recommended setting: 0 (unless you wish to try working on it).
```

> **wxUSE_COMPILER_TLS** [The comment](https://github.com/wxWidgets/wxWidgets/blob/8a77fae68b8758bbdcd729e5680e0c4bd874bfb6/include/wx/setup_inc.h#L284) mentions issues with Windows XP but since wxWidgets 3.3 supports only Vista+, not only should the comment be changed but wx/msw/chkconf.h should be modified as well?

We should drop this option and update the code to not use it any more, I've added this to #22689.

> **wxUSE_INTL**
> 
> > i18n support: _() macro, wxLocale class. Requires wxTextFile.
> 
> Does it really require wxTextFile - I cannot see that?

The only way to be really sure is to try building with `wxUSE_INTL==1` but `wxUSE_TEXTBUFFER==0`. But it will probably fail due to other reasons...

IOW I think the last part can indeed be removed but just because I'm almost certain that building with `wxUSE_TEXTBUFFER==0` must not work anyhow.

> **wxUSE_DIALUP_MANAGER**
> 
> > Requires wxUSE_DYNAMIC_LOADER.
> 
> This does not seem to be the case.

We probably should remove `wxUSE_DIALUP_MANAGER` completely...

> **wxUSE_DYNAMIC_LOADER**
> 
> > experimental, don't use for now
> 
> Is it still experimental? Is it even used in wxWidgets code? I found `dynload.h` included in several places (e.g.,` src\msw\gdiplus.cpp`) but AFAICT, the code uses just `wxDynamicLibrary` (`dynlib.h`) and nothing from `dynload.h`?

I'm almost sure nobody uses this one because it has never been documented or even demonstrated in a sample and so I'd be surprised if anybody really knew what it did (I think it tries to make wxRTTI in loadable plugins work, but I have no idea if it actually succeeds in doing it). AFAICS from a quick search, Audacity name drops it in their `ModuleManager.h` but doesn't reference it otherwise and nobody else really uses it neither.

So I think it should be deprecated too, unless somebody is motivated in reviving it and at least providing documentation and the tests.

> **wxUSE_XML**
> 
> > XML parsing classes. Note that their API will change in the future, so using wxXmlDocument and wxXmlNode in your app is not recommended.
> 
> Is the Note correct?

No, it probably won't change.

> **wxUSE_GEOMETRY**
> 
> > Miscellaneous geometry code: needed for Canvas library
> 
> Affine transformation seem to be used by `wx{GC}DC`? See also` wxUSE_DC_TRANSFORM_MATRIX` remark below.

We should just remove this option and always compile this in, there is relatively little code guarded by this option and nothing is gained by disabling it.

> **wxUSE_FILEDLG**
> 
> > TODO: setting to choose the generic or native one
> 
> Remove?

Yes.

> **wxUSE_IPC, wxUSE_HELP, wxUSE_CONSTRAINTS, and others** I wonder why the comments for these are written like that instead of above the define as usual?

Just because I haven't updated them when rewriting the descriptions of all the options in a more detailed way. I guess it should be done...

> **wxUSE_DC_TRANSFORM_MATRIX** It seems this is in source code used for detecting availability of `wxAffineMatrix2D`, but `wxAffineMatrix2D` depends on `wxUSE_GEOMETRY`?

Again, I didn't check it that carefully, but I think we should just remove this option and always compile the relevant code.

Thanks for looking at this!

vadz committed:
Don't install latest Python when running inside container

This doesn't work because the action installs an ABI-incompatible
version of Python 3.10, resulting in errors like the following

python3: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found
(required by /__t/Python/3.10.7/x64/lib/libpython3.10.so.1.0)

so try using the system version (3.6) to avoid this.

vadz committed:
Install xvfb when using a container

Unlike the full GitHub environment, this package is not pre-installed in
a "plain" Ubuntu container.

vadz committed:
Install the specified compiler package in container

Don't always install g++ as an existing build uses clang.

Also modify another build to use g++-4.8 as this is the oldest still
supported compiler and so it's nice to have a build using it.

vadz committed:
Avoid warnings about deprecated functions inside glib.h

This happens when using gcc 4.8 under Ubuntu 18.04 and has to be
suppressed in the source file including the header as just doing it in
the header itself, with the usual wxGCC_WARNING_SUPPRESS/RESTORE macros,
has no effect, i.e. the warning is still given even if we bracket glib.h
inclusion with them.

vadz committed:
Switch clang CI build to use Ubuntu 20.04

Running it in 18.04 container would require installing libc++-dev and
other packages, it's simpler to just write it on the GitHub-provided
runner where they're already installed.

Note that we need to explicitly choose to use clang-10, as the GitHub
runner has clang-11 and clang-12 installed as well and the default clang
version is 11, but only libc++-10-dev is installed, so using the default
clang with -stdlib=libc++ fails out of the box.

vadz committed:
Always set wxUSE_GRAPHICS_CONTEXT to 1 by default

Don't bother with compiler and gcc version check, all compilers should
provide GDI+ headers/libraries by now.

vadz committed:
Remove checks for gcc <= 4.8 and always assume it's 4.8+

Previous versions don't provide C++11 support which is required now
anyhow and so can't be used any longer.

vadz committed:
Add an explicit check for gcc version

This doesn't cost much and might be helpful to people accidentally using
an ancient compiler version.

vadz committed:
Stop using obsolete wxW64 macro

Still define it for compatibility, but at least don't use it in our own
code.

vadz committed:
Update wxMSW-specific options in wxUniv setup.h

This section of the file wasn't updated when rerunning update-setup-h,
add the missing line to it and commit the results of rerunning it.

vadz committed:
Remove WXARC_MEMBER_TEMPLATES and always use member templates

Member template functions are supported by all current compilers as they
are already used in other places without any problems, so there is no
need to test for this.

vadz committed:
Require MSVS 2015+ and remove checks for earlier versions

Simplify the code checking for _MSC_VER or __VISUALC__ or using
wxCHECK_VISUALC_VERSION() macro (which is not very useful any longer,
but keep it in case we need it again in the future).

vadz committed:
Remove HAVE_STRTOULL and HAVE_WCSTOULL

We can assume these functions are always available now as they are part
of C++11.

vadz committed:
Use wxVector::shrink_to_fit() directly

Don't use deprecated wxShrinkToFit() as now wxVector always has this
member function because it is available even in STL build when using
C++11.

vadz committed:
Remove MSVS 2005 and 2008 project files

These compilers are not supported any longer.

Also disable their regeneration in Bakefiles.bkgen.

vadz committed:
Rebake webview sample makefiles after recent changes

This corresponds to the changes done in samples/webview/webview.bkl in
0adf31169d (Add advanced handler to webview sample, 2022-09-14).

See #22797.

vadz committed:
Fix copy-paste errors in wxUILocale documentation

Simply remove the parts that made sense in wxLocale context, but not
here.

vadz committed:
Regenerate wxStyledTextCtrl documentation

Rerun src/stc/gen_iface.py to update the generated part of the interface
header.

This should have been part of 8ac10d28f8 (Fix all the other comments
with Doxygen grouping commands too, 2022-08-03) but was forgotten there.

See #22248.

vadz committed:
Use MSVS 2015 solutions for AppVeyor CI builds

These solutions work for MSVS 2017, 2019 and 2022 as well.

PBfordev commented:
I will update the comments as suggested.

I will also attempt to deprecate `wxUSE_GEOMETRY`, `wxUSE_DC_TRANSFORM_MATRIX`, `wxUSE_DIALUP_MANAGER`, and `wxUSE_DYNAMIC_LOADER`.

But I am not sure what exactly needs to be done:
1. Remove the guards from the source code.
2. Mark the defines as deprecated (e.g., "Deprecated: Not used by wxWidgets since version 3.3.0") in comments in `setup_inc.h` and in the define list in the docs.
3. Anything else?

It's probably best if I delay this at least until  #22854 is merged.

BTW, I assume there is no way I can suspend CI builds for this PR?

vadz commented:
> I will update the comments as suggested.
> 
> I will also attempt to deprecate `wxUSE_GEOMETRY`, `wxUSE_DC_TRANSFORM_MATRIX`, `wxUSE_DIALUP_MANAGER`, and `wxUSE_DYNAMIC_LOADER`.

Great, TIA!

> But I am not sure what exactly needs to be done:
> 
>     1. Remove the guards from the source code.
> 
>     2. Mark the defines as deprecated (e.g., "Deprecated: Not used by wxWidgets since version 3.3.0") in comments in `setup_inc.h` and in the define list in the docs.
> 
>     3. Anything else?

I think we may just remove all mentions of them, as this is not going to break anything. Am I missing something here?

> It's probably best if I delay this at least until #22854 is merged.

I hope to do it relatively soon...

> BTW, I assume there is no way I can suspend CI builds for this PR?

Removing all the files in `.github/workflows` (or maybe all except for the one checking for the code style and spelling, as this one can still be useful to run), `appveyor.yml`, `.cirrus.yml` and `.circleci/config.yml` will prevent the CI builds from running for this branch. Of course, this commit would need to be reverted later, when merging.

Using `[skip-ci]` in the commit message will disable most, but not all, of them too.

PBfordev commented:
Thanks for the information on how to easily temporarily suppress the CI.

Regarding the deprecated defines
> I think we may just remove all mentions of them, as this is not going to break anything. Am I missing something here?
I thought we must keep them both defined in `setup.h` and mentioned in the docs because we could not assume there were no users ever using these in their code. But I guess it safe to assume that due to the nature of those?

vadz commented:
You're right that somebody could possibly check for the values of these symbols in their code, even though I don't remember ever seeing this. But it doesn't cost much to add some "Deprecated options" section to `include/wx/setup_inc.h` and just define them all there with a comment saying that they're defined for compatibility only but that changing them is not supported.

Or we could just do it in `wx/platform.h` itself to avoid having these "options", which can't be changed, in a file which is supposed to be editable.

vadz committed:
Require C++11 in the minimal sample CMakefile too

We could, and probably should, set target-level property on wx libraries
instead, but we'll have to increase the minimum required CMake version
to 3.8 in order to be able to do it.

vadz committed:
Remove our own implementation of std::is_convertible

Just use the standard trait instead.

vadz committed:
Try ignoring the first test failure in clang/Ubuntu 20.04 build

For unknown reason this test fails when run for the first time on this
system, but succeeds during the subsequent runs.

vadz committed:
Merge branch 'ci-use-containers'

Run Ubuntu 18.04 builds in a container on GitHub CI to avoid depending
on the existence of a deprecated and soon to be discontinued runner
environment.

See #22856.

vadz committed:
Use v3 of checkout action in GitHub CI builds

The previously used v2 uses node12 which is being deprecated by GitHub
and won't be any longer available soon, so switch to the currently
supported version.

vadz committed:
Remove custom wxVector implementation

Always use std::vector<>, we don't support platforms not providing the
standard library classes any longer.

vadz committed:
Use std::wstring unconditionally

Don't test for its existence in configure and CMake and don't use
wxStdWideString in the code.

vadz committed:
Use std::string::compare() unconditionally

Don't test for it in configure/CMake and just always use it.

vadz committed:
Use <type_traits> unconditionally

Don't use or even mention tr1/type_traits any more.

vadz committed:
Stop testing for pre-standard hash map/set classes

Always use std::unordered_{map,set} when we use them at all, don't
bother with checking for their availability and with tr1 or even older
alternatives.

vadz committed:
Remove wxVector unit test

Testing this class doesn't make much sense any more as it's just a
typedef for std::vector and we can presumably rely on the standard class
working as expected.

PBfordev committed:
Prune mentions of no longer supported compilers from the docs

Remove mentions of older MSVC versions and very old gcc versions.

PBfordev committed:
Update MSVS debug visualisers

Remove autoexp.inc, this format is no longer supported by MSVC.

Merge wxWidgets.2013.natvis with wxWidgets.natvis.

Remove wxVector<> visualizer as it's not used any more and is just a
synonym for std::vector<> now.

vadz committed:
Give more details in wxMSVC_VERSION_AUTO documentation

Explicitly list all the prefixes used with it currently.

PBfordev committed:
Replace wxOVERRIDE and wxNOEXCEPT with override and noexcept

Don't use deprecated macros in wxWidgets itself.

vadz committed:
Update documentation of deprecated C++11-related macros

Still document them for completeness, but mention that they are
deprecated and shouldn't be used.

vadz committed:
Merge branch 'msw-double-buffer'

Enable double buffering by default for all windows in wxMSW.

See #22851.

vadz committed:
Merge branch 'cxx11'

Require C++11 compiler and start using some C++ features in the code.

See #22684, #22854.

vadz committed:
Use wxSTATEMENT_MACRO_BEGIN/END in wx/catch_cppunit.h

This is more clear and doesn't rely on internal Catch::alwaysFalse()
which doesn't exist in Catch v2.x any more.

vadz committed:
Stop using Catch::toString()

This function doesn't exist in Catch2 any longer, so prepare for the
upgrade by stopping to use it and converting things to string more
explicitly.

No real changes yet.

vadz committed:
Update to Catch v2

Use Catch2 public header instead of using internal Catch v1 headers: we
can do this now because Catch2 supports (albeit in a rather ugly way)
being used with PCH.

vadz committed:
Fix harmless signed/unsigned comparison warning in wxFile test

This warning didn't happen with Catch 1.x, but is given with 2.x, so
avoid it by casting to the correct type.

vadz committed:
Don't compare wide strings with narrow ones in the tests

This works with Catch 1.x, but not with Catch2, which seems to behave
more closely to the "real" comparisons.

vadz committed:
Avoid harmless -Wunused-variable in unit tests

This warning used to be suppressed in Catch headers, but this is not the
case any longer, so work around it in the tests themselves now.

vadz committed:
Avoid another warning triggered by the last warning workaround

Initialize a variable to avoid -Wmaybe-uninitialized triggered by using
wxUnusedVar() with it now.

lbartoletti committed:
Fix compilation when including wx/propgrid/editors.h early

This header uses wxBitmapBundle but didn't declare it, resulting in
compilation errors if wx/bmpbndl.h hadn't been included before it.

Add a forward declaration of wxBitmapBundle to this header to fix this.

Closes #22875.

vadz committed:
Replace a reference to a very old SourceForge ticket number

Not sure if it's the exact same ticket or if it's even relevant 18 years
later, but it seems to be close.

vadz committed:
Stop immediately updating wxAuiToolBar in its wxEVT_SIZE handler

It seems wrong to request an immediate update here and this results in a
problem since the changes of d4ced54755 (Merge branch
'msw-double-buffer', 2022-10-15) as OnPaint() was called before the
background style was set to wxBG_STYLE_PAINT in Create(), resulting in a
fatal assert failure.

See #22876.

vadz committed:
Merge branch 'linux-stack-walk'

Fix line numbers for the frames not in the main process in wxStackWalker
under Linux.

See #22873.

vadz committed:
Document C++11 requirement in the main README

This should have been part of ecf321924e (Merge branch 'cxx11',
2022-10-15), see #22854.

vadz committed:
Use nullptr instead of NULL in the code and documentation

This is a combination of running clang-tidy with modernize-use-nullptr
check for some ports (GTK, X11, OSX) and manual changes to the ports for
which it couldn't be used easily (MSW, DFB) and also manually updating
the docs.

Also replace NULL with null or nullptr in the comments as this is more
consistent with the use of nullptr in the code and makes it simpler to
grep for the remaining occurrences of NULL itself.

And also use null in the assert messages.

Only a few occurrences of "NULL" are still left in non-C files, mostly
corresponding to unclear comments or string output which it might not be
safe to change.

vadz committed:
Remove support for wxMotif

This port hasn't been updated for ages and is not used by anybody any
longer, so remove its code to facilitate maintenance.

Also remove references to this port from the documentation and most of
the other places (VMS-specific descrip.vms files still check for it
because it's not clear how to update them all), including configure.

Regenerate the latter and rebake all makefiles.

Finally document that this port is not available any longer.

vadz committed:
Remove a very obsolete and unused "publicity" document

Its contents has become embarrassingly obsolete by now.

gerstrong committed:
Render icon correctly in high DPI in wxGenericTreeCtrl

Retrieve the icon of the size corresponding to the current DPI from
wxBitmapBundle instead of always scaling up the icon with the default
size.

Closes #22879.

vadz committed:
Remove wxGTK1 port and GPE support

Remove the port files and references to it from the common headers and
elsewhere.

Also remove GPE (GNOME PDA Environment) support as libgpewidget is
unmaintained since 2006 or so and has never been really used.

Use __WXGTK__ to test for any version of wxGTK now. Still define
__WXGTK20__ for compatibility, but always define it now and don't test
for it in the library code.

vadz committed:
Add wxWindow::MSWAfterReparent() virtual function

Refactor the existing code to extract the part of Reparent() updating
the new parent style to an overridable virtual function.

No real changes yet.

vadz committed:
Fix drawing wxListCtrl since enabling double buffering by default

We need to avoid using WS_EX_COMPOSITED for any of the parents of this
native control and not just for the control itself.

vadz committed:
Enable -Wzero-as-null-pointer-constant in wx headers

This warning shouldn't need to be disabled any longer now that we use
nullptr instead of NULL.

vadz committed:
Move wxListCtrl::MSWResetParentComposited() to wxWindow

This function will need to be used from other places too, so make it
available in the base class itself.

Also disable WS_EX_COMPOSITED for this window itself too.

vadz committed:
Disable compositing when using wxDragImage

ImageList_BeginDrag() doesn't seem to be compatible with the use of
WS_EX_COMPOSITED, so turn it off when using it.

We probably could turn it back on again in EndDrag(), but for now keep
things simple and just fix the regression which resulted in wxDragImage
not working at all any more since d4ced54755 (Merge branch
'msw-double-buffer', 2022-10-15).

vadz committed:
Disable use of composition in forty demo under MSW

This is ugly but allows the carts to be visible while they're being
dragged again under MSW.

The game still doesn't work at all under GTK/Wayland (and probably
macOS).

vadz committed:
Document WS_EX_COMPOSITED impact on wxClientDC

This is a big and incompatible change but, arguably, is still worth
making because it makes wxMSW behaviour similar to that of wxOSX and
wxGTK when using Wayland.

Do document it prominently and also document MSWDisableComposited() as
it will probably end up being used in quite a lot of existing code.

vadz committed:
Remove wxQSORT_DECLARED hack

This is not needed any longer since we don't need wxQsort() in
wx/vector.h any more after eb97bf90e4 (Remove custom wxVector
implementation, 2022-10-11).

vadz committed:
Remove documentation of wxPostDelete()

This function not only doesn't exist, but seems to never have existed at
all.

Not really sure what happened here, but we definitely don't need to
document it any more.

vadz committed:
Remove commented out code from wxGTK wxDataViewCtrl

This code has been unused since 2009, so it's probably not very useful
to keep it there.

No real changes.

vadz committed:
Deprecate wxQsort()

This function is not really useful and doesn't seem to be used anywhere.

vadz committed:
Don't use wxOVERRIDE in generated wxSTC files

This should have been done together with 88d526660f (Replace wxOVERRIDE
and wxNOEXCEPT with override and noexcept, 2022-10-12).

vadz committed:
Merge branch 'deprecate-wxqsort'

Deprecate never used wxQsort() function.

See #22884.

vadz committed:
Merge branch 'msw-composited-fixes'

Fix regressions in wxMSW after enabling WS_EX_COMPOSITED and document
this change.

See #22882.

vadz committed:
Replace wxOVERRIDE with override in Objective-C++ files too

This completes the changes of 88d526660f (Replace wxOVERRIDE and
wxNOEXCEPT with override and noexcept, 2022-10-12).

vadz committed:
Add check for obsolete macros to automatic code style checks

Avoid checking in code using wxOVERRIDE or wxNOEXCEPT again, now that
we've got rid of them.

vadz committed:
Add helper wxWithImages::GetImageBitmapFor()

This function is simpler to use than calling GetBitmapBundle() first and
then calling GetBitmapFor() on it, avoids creating a temporary
wxBitmapBundle unnecessarily when wxImageList is used and, most
importantly, is public and not protected, which will allow
wxGenericTreeCtrl to use it for the other image lists than the main one.

The slightly unwieldy name is due to trying to use "Image" in all public
functions of this class, which should hopefully minimize clashes with
any user-defined functions that may exist in the classes deriving from
it, without requiring the use of even uglier "WX" prefix.

No real changes yet.

vadz committed:
Support high DPI state, button bitmaps in wxGenericTreeCtrl too

This is similar to the grandparent commit for the normal bitmaps and
uses the version of the icon appropriate for the current DPI instead of
always scaling up the normal-sized image.

vadz committed:
Remove commented out code using GetImageList()

No real changes, just remove final mentions of GetImageList() as this
function shouldn't be used at all -- and now is not, any longer.

vadz committed:
Add wxWithImages::GetImageLogicalSize() helper too

This is similar to the just added GetImageBitmapFor() and works both
when using legacy wxImageList and wxBitmapBundle.

Use the new function in wxGenericTreeCtrl code.

vadz committed:
Check for any images presence in wxGenericTreeCtrl

Use HasImages() instead of GetImageList() as the goal here is to check
if we have any images and not whether we specifically have an image
list.

vadz committed:
Use wxDCClipper instead of manual DestroyClippingRegion()

No real changes, just avoid manual cleanup and use a RAII helper
instead.

vadz committed:
Merge branch 'remove-motif-and-gtk1'

Remove obsolete and unmaintained wxMotif and wxGTK1 ports.

See #22877.

vadz committed:
Merge branch 'nullptr'

Use nullptr instead of NULL in the code, comments and the documentation.

See #22881.

vadz committed:
Check for adding NULL macro too

This might be too coarse, e.g. NULL is still needed in C code, so this
might need to be restricted to C++ files only later.

vadz committed:
Switch CMake wxGTK build to Ubuntu 22.04

Avoid warnings about using deprecated Ubuntu 18.04 runner.

vadz committed:
Avoid deprecation warnings for gha-setup-vsdevenv action

Switch to a local fork of the action using node 16 and latest
actions/core package as both are necessary to avoid getting deprecation
warnings.

vadz committed:
Update microsoft/setup-msbuild to avoid deprecation warnings

Upgrade it to v1.1.3.

vadz committed:
Update hendrikmuhs/ccache-action to avoid deprecation warnings

Upgrade it to v1.2.3.

vadz committed:
Merge branch 'generic-treectrl-hidpi'

Generic wxTreeCtrl icons high DPI improvements.

See #22887.

vadz committed:
Merge branch 'catch-2.x'

Switch to using CATCH v2 for the tests.

See #22874.

vadz committed:
Merge branch 'ci-avoid-deprec'

Fix deprecation warnings in GitHub Actions CI builds.

See #22888.

vadz committed:
Correct wxBitmapBundle best match choice description in its docs

Closes #22891.

vadz committed:
Fix wxTaskBarButton availability in the documentation

It's only available under MSW and doesn't have a generic implementation.

vadz committed:
Fix MSW "Save" dialog overwrite prompt for files without ext

We need to call IFileDialog::SetDefaultExtension() to ensure that the
native dialog itself appends the extension itself to the files entered
without extension it, otherwise it doesn't do it at all, in spite of
SetFileTypes() being already called, and so doesn't show "Confirm
overwrite" prompt for them, while wxMSW own code does append the
extension later, resulting in the existing files being overwritten
without any confirmation.

It would probably be a good idea to stop appending the extension on our
own and just use the one appended by the dialog to ensure that we don't
have similar problems in the future, but don't change this yet.

Closes #22898.

PBfordev committed:
Temporarily disable CI builds: Will be reverted!

PBfordev committed:
More comment updates

PBfordev committed:
Merge branch 'update-setuph-comments' of https://github.com/PBfordev/wxWidgets into update-setuph-comments

PBfordev commented:
I have updated more comments and I may be done with the PR. I have still temporarily disabled CI, in case changes will be requested.

I was thinking about this and the rest of changes discussed here (e.g., deprecating some `wxUSE_xxx`) should be done in a separate PR, as it will also affect the code (the guards will be removed). This PR should really do only what is says, i.e., just update the comments.

Based on recent changes, the comments and default values for standard classes use defines are probably going to change as well, so that would be likely another PR.

vadz commented:
Thanks, I'm going to push this soon. I'm not sure if this should be backported to 3.2 branch: AFAICS everything here except for `WXWIN_COMPATIBILITY_3_0` comment is valid for it too, but I could be missing something, please let me know if I do.

FWIW I could cherry pick the commits from your branch without any problems (thinking about it again, I could have used `git merge --no-commit` to do it even simpler), but to avoid such problems in the future, please consider rebasing, i.e. running `git rebase master <your-branch>`, rather than merging master into your branch -- this is what you typically want to do if you need to integrate the changes on master into your working branch (but for short-lived branches you typically don't need to do neither the former not the latter, in fact).

PBfordev commented:
I can't see anything that could not be backported, aside for the already-mentioned `WXWIN_COMPATIBILITY_3_0` comment. But as I wrote before, I think the comments for both `WXWIN_COMPATIBILITY_2_8` and `WXWIN_COMPATIBILITY_3_0` are already wrong there as they are identical, i.e., similar issue as here.

`WXWIN_COMPATIBILITY_2_8` comment claims that the default is `1` while it actually is `0`. The wxWidgets versions where the two compatibilities are going to be removed should be different as well.

FWIW, I thought I did use rebase on my PR branch (with TortoiseGIT via its Fetch & Rebase), apparently not.
